services/workflowbuilder/src/infra/http/controllers/procedureProjects.controllers.ts

```diff
@@ -1,34 +1,72 @@
 import { procedureProjectMapper } from '../mappers/proccedureProject-mapper';
 import { ApiResponse } from '@nestjs/swagger';
+import { ProcedureProject } from 'src/application/entities/procedureProject';
+import { PrismaClient } from '@prisma/client';
+import { prismaProcedureProjectRepository } from 'src/infra/database/prisma/repositories/prisma-procedureProject-repository';
 
 @Controller()
-export class procedureProjectController{
-    constructor(
-        private CreateProcedureProjectUseCase: createProcedureProjectUseCase
-    ){}
+export class procedureProjectController {
+  constructor(
+    private CreateProcedureProjectUseCase: createProcedureProjectUseCase,
+    private prismaProcedureProjectRepository: prismaProcedureProjectRepository,
+  ) {}
 
-    @Post()
+  @Post()
+  @ApiResponse({
+    status: 201,
+    description: 'The WorkFlowBuilder has been successfully created.',
+  })
+  @ApiResponse({
+    status: 500,
+    description: 'Internal server error.',
+  })
+  async create(@Body() body: createProcedureProjectDto) {
+    const { name, machineName, description } = body;
+    const { procedureProject } =
+      await this.CreateProcedureProjectUseCase.execute({
+        name,
+        machineName,
+        description,
+      });
+    return { procedureProject: procedureProjectMapper.toDto(procedureProject) };
+  }
 
-    @ApiResponse({
-        status: 201,
-        description: 'The WorkFlowBuilder has been successfully created.',
-      })
-      @ApiResponse({
-        status: 500,
-        description: 'Internal server error.',
-      })
-    async create(@Body() body: createProcedureProjectDto){
-        const { name, machineName, description} = body;
-        console.log(machineName);
-        
-        const {procedureProject} = await this.CreateProcedureProjectUseCase.execute({
-            name,
-            machineName,
-            description
-        });
-    return {procedureProject: procedureProjectMapper.toDto(procedureProject)}
+  @Get('search/:field/:value')
+  @ApiResponse({
+    status: 200,
+    description:
+      'All the "ProcedureProject" records with matching fields were found successfully.',
+  })
+  @ApiResponse({
+    status: 500,
+    description: 'Internal server error.',
+  })
+  async findByField(
+    @Param('field') field: string,
+    @Param('value') value: string,
+  ): Promise<ProcedureProject[] | null> {
+    const procedureProject  = await this.prismaProcedureProjectRepository.findByField(field,value)
+    return procedureProject
+  } 
+
+  @Get()
+  @ApiResponse({
+    status: 200,
+    description: 'All the "ProcedureProject" had been found',
+  })
+  @ApiResponse({
+    status: 500,
+    description: 'Internal server error.',
+  })
+  async findAll(): Promise<ProcedureProject[] | null> {
+    const procedureProjects =
+      await this.prismaProcedureProjectRepository.findAll();
+    if (!procedureProjects) {
+      console.log('No "ProcedureProject" were found');
```

Comment:
Here, you need to throw a not success response https://docs.nestjs.com/exception-filters#throwing-standard-exceptions